src/cache/cacher.rs

```diff
@@ -69,6 +70,51 @@ pub trait Cacher: Send + Sync {
     fn hash_url(&self, url: &str) -> String {
         blake3::hash(url.as_bytes()).to_string()
     }
+    
+    /// A helper function that returns either compressed or encryption results
+    /// Feature flags are required  for this to work
```

Comment:
```suggestion
    /// A helper function that returns either compressed or encryption results
    /// Feature flags are required  for this to work
    ///
    /// # Arguments 
    /// 
    /// * `search_results` - It takes the json search results string as an argument.
    /// * `url` - It takes the url of the queried pages as an argument.
    ///
    /// # Error
    /// Returns the compressed or encrypted bytes on success otherwise it returns a CacheError
    /// on failure.
```

Sorry for the delay in review. 
For the documentation around the functions, structs and enums, I would suggest following the rust convention for this. 

Here is the link to the rust convention on writing documentation around code:
- https://rust-lang.github.io/api-guidelines/documentation.html

Also, for more examples on how to write documentation around code, I would recommend reviewing other functions, structs and enums in the project. :slightly_smiling_face: